src/runtime/hack/fc_metrics.py

```diff
@@ -0,0 +1,282 @@
+#
+# Copyright (c) 2020 Ant Financial
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# This file is formatted by https://github.com/google/yapf
+#
+
+import sys
+import requests
+import json
+
+
+def underbar_to_camel(s):
+    ss = s.split("_")
+    r = ""
+    for s in ss:
+        r = r + s.capitalize()
+    return r
+
+
+def json_tag(s):
+    return "`json:\"" + s + "\"`"
+
+
+metric_declare_stmt = """{fn} = prometheus.NewGaugeVec(prometheus.GaugeOpts{{
+        Namespace: fcMetricsNS,
+        Name:      "{tag}",
+        Help:      "{help}",
+    }},
+        []string{{"item"}},
+    )"""
+
+
+def lowercase(s):
+    # RTCDeviceMetrics need special rule
+    if s == "RTCDeviceMetrics":
+        return "rtcDeviceMetrics"
+    return s[0].lower() + s[1:]
+
+
+def uppercase(s):
+    # RTCDeviceMetrics need special rule
+    if s == "rtcDeviceMetrics":
+        return "RTCDeviceMetrics"
+    return s[0].upper() + s[1:]
+
+
+def generate_declaration(fn, tag, help):
+    fn = lowercase(fn)
+    return metric_declare_stmt.format(fn=fn, tag=tag, help=help)
+
+
+type_name_tag_map = {}
+
+# only for FirecrackerMetrics struct
+type_name_field_name_map = {}
+# all metrics list
+all_metrics = []
+# all lines that used to definition go struct.
+go_types = []
+
+
+def parse_metric_declaration(lines):
+    # help: name: tag: items:{go_var: json_tag}
+    metric = {}
+    in_struct = False
+    in_FirecrackerMetrics = False
+    line_count = len(lines)
+    i = 0
+    while i < line_count:
+        line = lines[i]
+        i = i + 1
+
+        line = line.rstrip()
+        lline = line.lstrip()
+        if lline.find("#[") > -1:
+            continue
+        if line == "struct SerializeToUtcTimestampMs;":
```

Comment:
Wouldn't it be safer to skip lines matching `^struct [^ ]*;$` or similar to avoid having to hard-code this name?